hera_cal/utils.py

```diff
@@ -1397,10 +1485,232 @@ def red_average(data, reds=None, bl_tol=1.0, inplace=False,
+    list, and match them to the LST-edges, this is slow for a large list of files,
+    mostly because of the time it takes to open each file (not read it!). 
+
+    To speed this up, this function does two things: (1) it uses filenames to get the
+    first time in each file, if possible. This makes it *approximate*, not exact, but 
+    this quickly whittles down the number of files that need to be opened. (2) It
+    uses bounded binary search to find the files that have times in the given full
+    LST range, with an approximate knowledge of how many files to move forward or
+    backward after each file read, based on the integration time. This means many
+    fewer files need to be opened (this is much faster if the files themselves describe
+    a regular grid of JDs, but works either way due to the binary search). This latter
+    speedup is really only a speed up if the filename-time matching is not available.
+
+    If the filename-matching is not available, and the LST-edges cover a full 2pi
+    steradians, then this function is as slow (or perhaps a bit slower) than something
+    like ``config_lst_bin_files``. 
+
+    Parameters
+    ----------
+    lst_range
+        The lst range to match. This is a tuple of (start, stop) lsts in radians.
+    file_list
+        A list of files to match against. This can be a list of strings, a list of
+        pathlib.Path objects, or a list of FastUVH5Meta objects.
+    files_sorted
+        If True, assume that the file_list is already sorted by increasing time. 
+        If False, sort the file_list by time within the function, using ``sort_fn``.
+    sort_fn
+        The function to use to sort the file_list in increasing time. This should be a 
+        function that takes a list of str or Path objects and returns a list of the same
+        in order.
+    jd_regex
+        The regex to use to extract the julian date from the file name. If provided,
+    approximate
+        If True, return a list of files that is a superset of those that exist in the
+        LST bins -- either the exact list or more than the true list. This will be based
+        on times inferred from the filename (if possible) and the integration time of
+        the first file. 
+    
+    Returns
+    -------
+    list of FastUVH5Meta
+        The list of FastUVH5Meta file objects that have lsts that fall into the given
+        LST range.
+    """
+    if not files_sorted:
+        file_list = sort_fn(file_list)
+
+    # Get performance parameters from the first file in the list
+    meta = FastUVH5Meta(
+        file_list[0], 
+        blts_are_rectangular=blts_are_rectangular, 
+        time_axis_faster_than_bls=time_axis_faster_than_bls
+    )
+    if blts_are_rectangular is None:
+        blts_are_rectangular = meta.blts_are_rectangular
+    if time_axis_faster_than_bls is None:
+        time_axis_faster_than_bls = meta.time_axis_faster_than_bls
+
+    file_list = [
```

Comment:
I find it confusing that this is also called `file_list`... it's really `metadata_list` or something similar.

Reply:
OK, changed to `metadata_list`.